tools/GulpUtils.ts

```diff
@@ -19,51 +19,67 @@ import os = require ("os");
 import path = require ("path");
 import Q = require ("q");
 import util = require ("util");
-import zlib = require ("zlib")
+import zlib = require("zlib");
 
 class GulpUtils {
     private static TestCommand: string = "test";
+    private static CoverageJsonFileName: string = "coverage.json";
 
-    public static runAllTests(modulesToTest: string[], modulesRoot: string): Q.Promise<any> {
-        return modulesToTest.reduce(function (soFar: Q.Promise<any>, val: string): Q.Promise<any> {
-            return soFar.then(function (): Q.Promise<any> {
+    public static runCoverage(modulesToTest: string[], modulesRoot: string, coverageResultsPath: string): Q.Promise<any> {
+        coverageResultsPath = path.resolve(coverageResultsPath);
+        var coverageJsonFiles: string[] = [];
 
-                var modulePath = path.resolve(modulesRoot, val);
-                // check if package has any tests
-                var pkg = require(path.join(modulePath, "package.json"));
-                if (!pkg.scripts || !(GulpUtils.TestCommand in pkg.scripts)) {
-                    return Q({});
-                }
+        // for each module, sequentially run tests with coverage (with json reporting)  
+        return modulesToTest.reduce(function(soFar: Q.Promise<any>, val: string): Q.Promise<any> {
```

Comment:
Is there any way to make our tests run in parallel? That would make the run time a lot shorter.